test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/lang/ClassJSTest.scala

```diff
@@ -0,0 +1,36 @@
+/*                     __                                               *\
+**     ________ ___   / /  ___      __ ____  Scala.js Test Suite        **
+**    / __/ __// _ | / /  / _ | __ / // __/  (c) 2013, LAMP/EPFL        **
+**  __\ \/ /__/ __ |/ /__/ __ |/_// /_\ \    http://scala-js.org/       **
+** /____/\___/_/ |_/____/_/ | |__/ /____/                               **
+**                          |/____/                                     **
+\*                                                                      */
+package org.scalajs.testsuite.javalib.lang
+
+import org.junit.Test
+import org.junit.Assert._
+
+import scala.scalajs.js
+
+class ClassJSTest {
+  @noinline
+  def testNoInline(clazz: Class[_], componentType: Class[_]): Unit =
+    assertEquals(componentType, clazz.getComponentType)
+
+  @inline
+  def test(clazz: Class[_], componentType: Class[_]): Unit = {
+    testNoInline(clazz, componentType)
+    assertEquals(componentType, clazz.getComponentType)
+  }
```

Comment:
These should be local defs in `getComponentType()` (I thought I said that before, but apparently my comment disappeared ^^)